### PR TITLE
Add the thread state description in 2.2

### LINKED ISSUE #1865: Context "control thread state" and "rendering thread state"
In many of the algorithms related to creating/suspending/resuming a BaseAudioContext, references are made to "control thread state" and "rendering thread state".  Perhaps instead of write `<code>control thread state</code>`, we should make these internal slots of the context?

But perhaps it's too late to do this for V1?

### PR DESCRIPTION
**Pull request by hoch**
Fixes #1865.

- Define `control thread state` and `rendering thread state` in the graph rendering section.

***
<a href="https://pr-preview.s3.amazonaws.com/hoch/web-audio-api/pull/1918.html" title="Last updated on May 23, 2019, 2:52 PM UTC (c334da4)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WebAudio/web-audio-api/1918/85a872c...hoch:c334da4.html" title="Last updated on May 23, 2019, 2:52 PM UTC (c334da4)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -10701,6 +10701,11 @@ real-time, callback-based audio thread, if computing audio for an
 {{AudioContext}}, or a normal thread if rendering and audio graph
 offline using an {{OfflineAudioContext}}.
 
+Each thread has an internal slot indicates its current state.
+<dfn>control thread state</dfn> is the equivalent of {{BaseAudioContext/state}}
```
Comment (rtoy) [inline index.bs]: Notationally, we've used `[[foo]]` to mean an internal slot of an object. I don't think we're totally consistent with this usage, but I think we've tried to update the text to use this.

Comment (hoch) [reply index.bs]: I am not sure about this one. `control thread state` is basically `BaseAudioContext.state` so technically it doesn't need another internal slot for it. That's why I chose this pattern (more loosely defined for the implementation detail).

Comment (hoch) [reply index.bs]: I've added a comment to the definition of `BaseAudioContext.state`.

File: index.bs [hunk 2]
```diff
@@ -2073,12 +2073,21 @@ Methods</h4>
 
 			1. Abort these steps and reject <var>promise</var> with
 				{{InvalidStateError}} when any of following conditions is true:
-				- The <em>control thread state</em> flag on the {{OfflineAudioContext}}
+				- The <a>control thread state</a> on the {{OfflineAudioContext}}
 					is <code>closed</code>.
 				- The {{[[rendering started]]}} slot on the {{OfflineAudioContext}}
 					is <em>false</em>.
 
+<<<<<<< HEAD
```
Comment (rtoy) [inline index.bs]: Don't forget to fix this conflict!

File: index.bs [hunk 3]
```diff
@@ -2073,12 +2073,17 @@ Methods</h4>
...
 			1. Abort these steps and reject <var>promise</var> with
 				{{InvalidStateError}} when any of following conditions is true:
-				- The <em>control thread state</em> flag on the {{OfflineAudioContext}}
+				- The <a>control thread state</a> on the {{OfflineAudioContext}}
 					is <code>closed</code>.
 				- The {{[[rendering started]]}} slot on the {{OfflineAudioContext}}
 					is <em>false</em>.
 
-			1. Set the <em>control thread state</em> flag on the
+			If the <a>control thread state</a> flag on the
```
Comment (rtoy) [inline index.bs]: This is indented incorrectly somehow.  Wasn't this removed in PR #1887

Comment (hoch) [reply index.bs]: I am not sure how this happened. Removed the paragraph again.

File: index.bs [hunk 4]
```diff
@@ -10679,6 +10679,11 @@ reaction to the calls from the <a>control thread</a>. It can be a
 real-time, callback-based audio thread, if computing audio for an
 {{AudioContext}}, or a normal thread if computing audio for an {{OfflineAudioContext}}.
 
+Each thread has an internal slot indicates its current state.
```
Comment (rtoy) [inline index.bs]: Typo:  "slot indicates" -> "slot that indicates"

**Pull request merged**